ads/fusion.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function fusion(global, data) {
+  validateData(data, [], ['mediaZone', 'layout', 'adServer', 'space']);
+
+  const container = global.document.getElementById('c');
+  const ad = global.document.createElement('div');
+  ad.setAttribute('data-fusion-space', data.space);
+  container.appendChild(ad);
+
+  writeScript(global, 'https://assets.adtomafusion.net/fusion/latest/fusion-amp.min.js', () => {
+    global.Fusion.apply(container, global.Fusion.loadAds(data));
+
+    global.Fusion.on.adResized.run(ev => {
+      if (ev.width >= 20 && ev.height >= 20) {
+        global.context.renderStart();
+      }
```

Comment:
@tomaszwroblewski in the case of passbacks, will you still be updated on the ad rendering status? If not I think you are correct about `renderStart` not applicable in your case. Because we would need all passback ad networks to support `renderStart` to make this correct. Otherwise, we would assume that `renderStart` is not received, and thus assume a timeout for the ad. Sorry about the confusion.